### ISSUE TITLE
Replace compression JNI with pure java

### ISSUE THREAD

**Issue by kirillp** [eclipse/packager#23]
In order to simplify the code, make it more secure and more portable across systems (coming RISC-V) 
It would be nice to replace JNI based compression to pure java

https://github.com/airlift/aircompressor

It also faster

**Comment by dwalluck** [comment]
The compression code uses pure Java via Apache Commons Compress. The only thing I can think of that you might be referring to is Zstandard which is provided via zstd-jni and is optional.

In that case, you should raise this issue with the Commons Compress project.

**Comment by ctron** [comment]
I agree it would be great to have pure Java support for Zstd as well. And the mentioned repository might be of help.

However, from what I saw (and maybe I overlooked something), it doesn't provide an `OutputStream` implementation. Which is something that we need. And as RPM writing/reading with Zstd on RISC-V seems a pretty niche use case, I don't see much value in adding a lot of code around this in this project.

As @dwalluck suggested, maybe reach out to the Apache Commons Compress project and have them extend their code.
